lib/ddtrace/tracer.rb

```diff
@@ -441,7 +444,8 @@ def activate_priority_sampling!(base_sampler = nil)
       @sampler = if base_sampler.is_a?(PrioritySampler)
                    base_sampler
                  else
-                   PrioritySampler.new(base_sampler: base_sampler)
+                   PrioritySampler.new(base_sampler: base_sampler,
+                                       post_sampler: Datadog::RateByServiceSampler.new(1.0, env: tags[:env]))
```

Comment:
why is this change needed?

Reply:
Because of https://github.com/DataDog/dd-trace-rb/pull/868/files#diff-075cdbe46a2ed147b75f5da176ae2e88L72

This is how you ensure env is still properly applied from the tracer to the sampler.

Reply:
what happens if the env tag is changed after we initialize a priority sampler?

Reply:
It won't be read; same as it is today.The only difference is maybe you were configuring tags at the same time you activated priority sampling manually, so maybe it would've applied, maybe it wouldn't; depends on the order the settings were applied.

Only conceivable way is to change the value to a lazy `Proc`. You could argue that the `key_for(span)` function should be this `Proc`, and when you define a `RateByServiceSampler`, you should provide this proc that acts as means of resolving a key. Which would make this more of a `RateByKeySampler` which can optionally configured to use an `env` or not, which I think would be a better design.

Reply:
Okay, I did a small refactor of the `RateByServiceSampler` and changed it to resolve the env from a `Proc`.